teps/0132-queueing-concurrent-runs.md

```diff
@@ -0,0 +1,313 @@
+---
+status: proposed
+title: Queueing Concurrent Runs
+creation-date: '2023-02-24'
+last-updated: '2023-02-24'
+authors:
+- '@lbernick'
+collaborators:
+- '@chengjoey'
+- '@jerop'
+---
+
+# TEP-0132: Queueing Concurrent Runs
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+  - [Existing Workarounds](#existing-workarounds)
+  - [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes and Caveats](#notes-and-caveats)
+- [Design Details](#design-details)
+- [Design Evaluation](#design-evaluation)
+  - [Reusability](#reusability)
+  - [Simplicity](#simplicity)
+  - [Flexibility](#flexibility)
+  - [User Experience](#user-experience)
+  - [Performance](#performance)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Implementation Plan](#implementation-plan)
+  - [Test Plan](#test-plan)
+  - [Infrastructure Needed](#infrastructure-needed)
+  - [Upgrade and Migration Strategy](#upgrade-and-migration-strategy)
+  - [Implementation Pull Requests](#implementation-pull-requests)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes allowing users to control the number of PipelineRuns and TaskRuns (and maybe CustomRuns) that can run concurrently.
+The focus of this TEP is different from [TEP-0120: Canceling Concurrent PipelineRuns](./0120-canceling-concurrent-pipelineruns.md),
+which focuses on PipelineRuns that may have ordering dependencies, but we may choose to develop a solution that addresses both TEPs.
+
+## Motivation
+
+[TEP-0092: TaskRun Timeouts](./0092-scheduling-timeout.md) proposed capping the amount of time a TaskRun could be queued for.
+However, the use cases specified in this proposal (running Tasks in a resource constrained environment, or "fire and forget")
+would also be met by queueing PipelineRuns or TaskRuns for execution. Queueing may also be easier to understand and more flexible.
+
+### Goals
+
+- Can "fire and forget" PipelineRuns by creating many PipelineRuns but preventing all of them from executing at once
+- Can control the number of matrixed TaskRuns that can execute at once
```

Comment:
TaskRuns, CustomRuns and PipelineRuns (coming soon through Pipelines in Pipelines) can be matrixed

```suggestion
- Can control the number of matrixed Runs that can execute at once
```